apps/api-service/src/transformer.ts

```diff
@@ -1,3 +1,90 @@
+import type { SuperJSONResult } from "superjson";
 import { SuperJSON } from "superjson";
 
-export const transformer = SuperJSON;
+/**
+ * Creates a transformer that wraps SuperJSON and warns about unsupported data types.
+ * Specifically, it detects class instances that haven't been registered with SuperJSON,
+ * as these cannot be properly serialized and may cause issues.
+ */
+function createWarningTransformer() {
+  const warnedTypes = new Set<string>();
+
+  /**
+   * Recursively checks for class instances in the data structure.
+   * Class instances are objects whose constructor is not one of the built-in types.
+   */
+  function detectUnsupportedTypes(value: unknown, path = "root"): void {
+    if (value === null || value === undefined) {
+      return;
+    }
+
+    // Handle arrays
+    if (Array.isArray(value)) {
+      value.forEach((item, index) =>
+        detectUnsupportedTypes(item, `${path}[${index}]`),
+      );
+      return;
+    }
+
+    // Only check objects (not primitives)
+    if (typeof value !== "object") {
+      return;
+    }
+
+    const constructor = value.constructor;
+    const constructorName = constructor?.name;
+
+    // Skip built-in types that SuperJSON handles natively
+    const builtInTypes = [
+      "Object",
+      "Array",
+      "Date",
+      "RegExp",
+      "Error",
+      "Map",
+      "Set",
+      "BigInt",
+      "URL",
+    ];
+
+    if (constructorName && !builtInTypes.includes(constructorName)) {
+      // This is a class instance - warn about it
+      const warningKey = `${constructorName}:${path}`;
+      if (!warnedTypes.has(warningKey)) {
+        warnedTypes.add(warningKey);
+        // oxlint-disable-next-line no-console
+        console.warn(
```

Comment:
don't use console.warn. throw an error instead. rename the function to assertNoUnsupportedTypes